src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -77176,6 +77175,39 @@ static void F()
                 Diagnostic(ErrorCode.ERR_NoExplicitConv, "(S?)F").WithArguments("method", "S?").WithLocation(6, 18));
         }
 
+        [WorkItem(33330, "https://github.com/dotnet/roslyn/issues/33330")]
+        [Fact]
+        public void NullableT_32()
+        {
+            var source =
+@"#nullable enable
+class Program
+{
+    static void F(int? i, int j)
+    {
+        if (i.HasValue)
+        {
+            _ = (int)(i & j);
+            _ = (int)(i | j);
+            _ = (int)(i ^ j);
+            _ = (int)(~i);
+        }
+    }
+    static void F(bool? i, bool b)
+    {
+        if (i.HasValue)
+        {
+            _ = (bool)(i & b);
+            _ = (bool)(i | b);
+            _ = (bool)(i ^ b);
+            _ = (bool)(!i);
+        }
+    }
+}";
```

Comment:
Consider including cases without `i.HasValue` to verify warnings are reported when necessary.

Reply:
Done.